contracts/sol6/token/KyberNetworkTokenV2.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity 0.6.6;
+
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+interface IERC20Burnable {
+    function burnFrom(address _from, uint256 _value) external returns (bool);
+}
+
+contract KyberNetworkTokenV2 is ERC20Burnable, Ownable {
+    using SafeERC20 for IERC20;
+
+    address public immutable oldKNC;
+    address public minter;
+
+    event Minted(address indexed account, uint256 indexed amount, address indexed minter);
+    event BurntAndMinted(address indexed account, uint256 indexed amount);
+    event MinterChanged(address indexed oldMinter, address indexed newMinter);
+
+    modifier onlyMinter() {
+        require(msg.sender == minter, "only minter");
+        _;
+    }
+
+    constructor(address _oldKNC, address _minter)
+        public ERC20("Kyber Network Crystal V2", "KNCv2")
+    {
+        require(_oldKNC != address(0), "invalid old knc");
+        require(_minter != address(0), "invalid minter");
+        oldKNC = _oldKNC;
+        minter = _minter;
+    }
+
+    function mint(address account, uint256 amount) external onlyMinter {
+        super._mint(account, amount);
+        emit Minted(account, amount, minter);
+    }
+
+    /// @dev burn old knc and mint new knc for msg.sender, ratio 1:1
+    function mintWithOldKnc(uint256 amount) external {
+        assert(IERC20Burnable(oldKNC).burnFrom(msg.sender, amount));
```

Comment:
oldKNC has 1 assert already, i feel it isn't really necessary to add another. Help save gas..? =)

Reply:
yeah it's true, actually it is either assertion or return true for burnFrom in old KNC contract haha